pkg/ruler/scheduler.go

```diff
@@ -287,7 +305,8 @@ func (s *scheduler) workItemDone(i workItem) {
 		level.Debug(util.Logger).Log("msg", "scheduler: stopping item", "user_id", i.userID, "group", i.groupName, "found", found, "len", len(currentRules))
 		return
 	}
-	next := i.Defer(s.evaluationInterval, i.groupName, currentRules)
+	// Can we ever get here with different rules? Shouldn't be possible based on the above if...
```

Comment:
I don't follow this question, why does the prior `if` preclude this call? (sorry if I'm just being dense)

Reply:
That was a note for me... I convinced myself that it is safe not to pass "currentRules" around. Deleting the comment now